src/global.ts

```diff
@@ -231,17 +247,28 @@ interface AuthenticationResult {
   error_description?: string;
 }
 
+interface TokenEndpointOptions {
+  baseUrl: string;
+  client_id: string;
+  grant_type: string;
```

Comment:
Could the token EP method switch the grant type based on the options type instead of declaring it specifically? Not wild about passing that in if we're already indicating in some way that an RT grant is being used.

Reply:
Right now the the token endpoint method doesn't exist inside the client so it doesn't have access to the options. We could pass those options in and perform the logic there, but I like that it serves as more of a utility function to simply call that token endpoint and pushes the logic up to the places that call it. Keeps it simpler, simpler testing, debugging, etc.